### PR TITLE
Add pages modal on summary page

### PR DESCRIPTION
**Pull request by piotrmark**
Changes:
- Removed unused code in SortableSelectionList
- Used isSummary prop to display plus icon on the sidebar
- Wrapped SelectPages site in modal

How to test:
- Go to summary page and the plus icon should trigger the modal with pages selection
- Changes done on modal should be reflected on the sidebar list
- Validation should work in the modal

image

### CODE REVIEW

**Review by streetcoder123 — DISMISSED** [review]

**Comment by annielovesjs** [comment]
When you click preview in the pages modal, it takes you to the preview page but it also display the modal weirdly

pagesErr

**Comment by xtinah-w** [comment]
Is it possible for us to just remove the "Preview" button just on this component so that we don't have a re-design the entire feature? This should just act as a lightweight adding tool for now.

Add Pages Modal

**Comment by piotrmark** [comment]
Actually the modal doesn't look good on really small screen (when cards stacks into 1 row), this has to be fixed before we merge that.

**Comment by jsondoo** [comment]
Taking this over from Piotr

1) Added repsonsiveness to the form
2) Removing preview button if it is on a modal

**Comment by jsondoo** [comment]
Expected Behavior:

You should be able to use the plus button to open the modal in the summary page (notice that preview link doesn't show up on the pages cards)
image

Pages selection part is responsive and scrollable (only when it's in a modal)
image

File: src/client/src/components/Modal/index.tsx [hunk 1]
```diff
@@ -20,6 +20,7 @@ import { MODAL_TYPES, ModalType } from "../../actions/modalActions/typeKeys";
 const getCustomStyles = (MODAL_TYPE: ModalType | undefined) => {
   // default width
   let CUSTOM_WIDTH = "50%";
+  let backgroundColor = "var(--vscode-menu-background)";
```
Comment (jasmingoh) [inline src/client/src/components/Modal/index.tsx]: use const isntead?

**Review by jasmingoh — APPROVED** [review]
works for me!
___
I noticed that when you add a blank page, the name of the 2nd blank page is "Blank" as well. Seems like @njindal239 already logged it as an issue #997

**Comment by annielovesjs** [comment]
@jsondoo  is this supposed to be noticeably  more bold than the other svgs
bold

**Review by streetcoder123 — APPROVED** [review]

**Comment by jsondoo** [comment]
@annielovesjs I'm just using the provided assets, not bolding anything

**Review by SahilTara — APPROVED** [review]

**Review by jsondoo — APPROVED** [review]

**Review by jcoc611-microsoft — APPROVED** [review]

**Pull request merged**